e2e/cypress/integration/team/teams_spec.js

```diff
@@ -182,4 +182,50 @@ describe('Teams Suite', () => {
         // * Ensure user is logged out
         cy.url({timeout: TIMEOUTS.LARGE}).should('include', 'login');
     });
+
+    it('TS14632 Team Icon should be Removed', () => {
+        cy.apiLogin('sysadmin');
+        cy.apiCreateTeam('test-team', 'Test Team').then(() => {
+            cy.visit('/');
+        });
+        cy.get('#sidebarHeaderDropdownButton').click();
+        cy.get('#teamSettings').click();
+
+        // 1. On a Team taht has a team icon
+        cy.get('#team_iconEdit').click();
+        cy.fileUpload('.section-max input', '../fixtures/mattermost-icon.png', 'image/png').trigger('change', {force: true});
+        cy.get('#saveSettingPicture').should('not.have.css', 'disabled');
+        cy.get('#saveSettingPicture').click();
+        cy.get('#teamSettingsModalLabel > .close').click();
+
+        // Making sure there is a team icon for active team
+        cy.get('.TeamIcon__image').should('be.visible');
```

Comment:
Please verify this within the context of team's button.  Like:
```javascript
cy.get(`#${team.name}TeamButton`).within(() => {
    cy.get('#teamIconImage').should('be.visible');
    cy.get('#teamIconInitial').should('not.exist');
});
```
Please add new IDs as [teamIconImage](https://github.com/mattermost/mattermost-webapp/blob/master/components/widgets/team_icon/team_icon.tsx#L49) and [teamIconInitial](https://github.com/mattermost/mattermost-webapp/blob/master/components/widgets/team_icon/team_icon.tsx#L57).